GP Mauna Loa with PyMC3 best practices (#316)
pymc-devs/pymc-examples: myst_nbs/gaussian_processes/GP-MaunaLoa.myst.md

    format_version: 0.13
    jupytext_version: 1.13.7
kernelspec:
  display_name: Python 3
  language: python
  name: python3
substitutions:
  extra_dependencies: bokeh
---

(GP-MaunaLoa)=
# Gaussian Process for CO2 at Mauna Loa

:::{post} April, 2022
:tags: gaussian process, CO2
:category: intermediate

> :author: Bill Engels

Reviewer: ```suggestion
:author: Bill Engels, Chris Fonnesbeck
```

Note: do the changes locally on the notebook, I comment here because I find it better to review.

## What day will the CO2 level break 400 ppm?

> How well do our forecasts look?  Clearly the observed data trends up and the seasonal effect is very pronounced.  Does our GP model capture this well enough to make reasonable extrapolations?  Our "training" set went up until the end of 2003, so we are going to predict from January 2004 out to the end of current year.  

Reviewer: ```suggestion
How well do our forecasts look?  Clearly the observed data trends up and the seasonal effect is very pronounced.  Does our GP model capture this well enough to make reasonable extrapolations?  Our "training" set went up until the end of 2003, so we are going to predict from January 2004 out to the end of 2022.  
```

The notebook should read correctly after 2022 finishes. Because it might not be updated before 2023 but also because we generate snapshots of the examples every few months so that books and the like can cite stable website and the snapshots could perfectly be read after 2022 and won't be updated, not even a comma

Next, we'll see about using PyMC3's GP functionality to improve the model, look at full posteriors, and incorporate other sources of data on drivers of CO$_2$ levels.

+++

## Authors
* Authored by Bill Engels in September, 2017 ([pymc#2444](https://github.com/pymc-devs/pymc/pull/2444))

> * Updated by Chris Fonnesbeck in December, 2020 ([pymc-examples#8](https://github.com/pymc-devs/pymc-examples/pull/8))

Reviewer: ```suggestion
* Updated by Chris Fonnesbeck in December, 2020
```

No need to add the PRs, and in this case the linked PR doesn't have any updates, it only restructures the untouched notebooks.
